### PR TITLE
docs: add Tree Icon prop

### LINKED ISSUE #25070: Add 'icon' prop to Tree API
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Each `TreeNode` within a `Tree` can specify an icon prop as a `ReactNode` or `Function(props)`.  Instead of having to specify a icon prop for every `TreeNode`, it would be much more efficient and easier if the `Tree` could also have an `icon` prop: `Function(props)`.  The `Tree` icon prop function would be used if the `Tree` has `showIcon=true` and if the `TreeNode` does not explicitly have an `icon` prop set.  Forcing each `TreeNode` to have a `icon` is redundant and inefficient in most cases.

### What does the proposed API look like?
```
<Tree 
    showIcon
    icon={(nodeProps) => { return (<div>Nice Icon</div>); }}
    treeData={treeData}/>
```

### PR DESCRIPTION
**Pull request by fireairforce**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
closes https://github.com/ant-design/ant-design/issues/25070

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    None.     |
| 🇨🇳 Chinese |    无。      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/tree/index.en-US.md](https://github.com/fireairforce/ant-design/blob/treeProps/components/tree/index.en-US.md)
[View rendered components/tree/index.zh-CN.md](https://github.com/fireairforce/ant-design/blob/treeProps/components/tree/index.zh-CN.md)

### CODE REVIEW

**Review by 0xPoe — APPROVED** [review]
LGTM，Thanks!

**Comment by xrkffgg** [comment]
新增的，发 feature 吧

**Comment by fireairforce** [comment]
> 新增的，发 feature 吧

不是新增的，本来就有这个api，加个文档说明

File: components/tree/index.en-US.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ Almost anything can be represented in a tree structure. Examples include directo
 | onLoad | Callback function for when a treeNode is loaded | function(loadedKeys, {event, node}) | - |  |
 | onRightClick | Callback function for when the user right clicks a treeNode | function({event, node}) | - |  |
 | onSelect | Callback function for when the user clicks a treeNode | function(selectedKeys, e:{selected: bool, selectedNodes, node, event}) | - |  |
+| icon | customize treeNode icon. | ReactNode/Function(props):ReactNode | - |  |
```
Comment (xrkffgg) [inline components/tree/index.en-US.md:55]:
```suggestion
| icon | customize treeNode icon. | ReactNode \| (props) => ReactNode | - |  |
```

Comment (fireairforce) [reply components/tree/index.en-US.md]: ok了

**Comment by xrkffgg** [comment]
https://ant.design/components/tree-cn/#TreeNode-props  这个中 不是有 icon 嘛？

**Comment by afc163** [comment]
image

不是 feature changelog 不能这么写。

**Comment by Rustin-Liu** [comment]
> 
> 
> https://ant.design/components/tree-cn/#TreeNode-props 这个中 不是有 icon 嘛？

这是节点的，不是tree的。

**Comment by fireairforce** [comment]
> https://ant.design/components/tree-cn/#TreeNode-props 这个中 不是有 icon 嘛？

顺手把这个里面的文档也调整了一下

File: components/tree/index.en-US.md [hunk 2]
```diff
@@ -60,7 +61,7 @@ Almost anything can be represented in a tree structure. Examples include directo
 | checkable | When Tree is checkable, set TreeNode display Checkbox or not | boolean | - |
 | disableCheckbox | Disables the checkbox of the treeNode | boolean | false |  |
 | disabled | Disables the treeNode | boolean | false |  |
-| icon | customize icon. When you pass component, whose render will receive full TreeNode props as component props | ReactNode/Function(props):ReactNode | - |  |
+| icon | customize icon. When you pass component, whose render will receive full TreeNode props as component props | ReactNode\| (props) => ReactNode | - |  |
```
Comment (xrkffgg) [inline components/tree/index.en-US.md:64]:
```suggestion
| icon | customize icon. When you pass component, whose render will receive full TreeNode props as component props | ReactNode \| (props) => ReactNode | - |  |
```

Comment (fireairforce) [reply components/tree/index.en-US.md]: fixed

**Comment by xrkffgg** [comment]
😅   我指出哪个，你就只改哪个啊，你好歹看看其他的呢、、、、

**Comment by fireairforce** [comment]
> 😅 我指出哪个，你就只改哪个啊，你好歹看看其他的呢、、、、

我错了😭 QuQ

**Pull request merged**